genapi/genapi.go

```diff
@@ -0,0 +1,361 @@
+// Package genapi implements a generic skeleton we can use as the basis for an
+// api service. It will set up command line arguments, connections to backend
+// databases, handle test modes which might affect those databases, and more.
+//
+// Basic definition
+//
+// To use first initialize a GenAPI instance somewhere accessible by your entire
+// application, and give it an RPC type
+//
+//	package myapi
+//
+//	var GA = genapi.GenAPI{
+//		Name:              "my-api",
+//		DefaultListenAddr: ":5678",
+//		RedisInfo: genapi.RedisInfo{
```

Comment:
Would this be better to be a pointer so we won't "use" it if its `nil` versus having to set `Use: true`?